lib/mongoid/extensions/time.rb

```diff
@@ -45,14 +45,18 @@ def configured
         #
         # @param [ Time ] object The time from Mongo.
         #
-        # @return [ Time ] The object as a date.
+        # @return [ Time | nil ] The object as a time.
         def demongoize(object)
           return nil if object.blank?
-          object = object.getlocal unless Mongoid::Config.use_utc?
-          if Mongoid::Config.use_activesupport_time_zone?
-            object = object.in_time_zone(Mongoid.time_zone)
+          if object.acts_like?(:time)
+            object = object.getlocal unless Mongoid::Config.use_utc?
+            if Mongoid::Config.use_activesupport_time_zone?
+              object = object.in_time_zone(Mongoid.time_zone)
+            end
+            object
+          elsif object.acts_like?(:date)
+            ::Date.demongoize(object).to_time
```

Comment:
MongoDB cannot store dates, it only stores times. When would this branch be taken exactly?

Reply:
That's a good point. The only situation I can think of is if the attributes hash is modified by the user to include a Date:
```
user = User.new
user.attributes["time"] = Date.new
user.time
```
Which perhaps is a little far fetched, but in the interest of exhaustiveness I feel like there's no reason to remove this. If you have a good reason to I will remove it.

Reply:
This needs to return Time or AS::TimeWithZone depending on how Mongoid is configured.

```
irb(main):025:0> a.t=Date.today
=> Wed, 13 Jul 2022
irb(main):026:0> a.t.class
=> ActiveSupport::TimeWithZone
irb(main):027:0> a.t
=> Wed, 13 Jul 2022 00:00:00 EDT -04:00
irb(main):028:0> a.attributes['t']=Date.today
=> Wed, 13 Jul 2022
irb(main):029:0> a.t
=> 2022-07-13 00:00:00 -0400
irb(main):030:0> a.t.class
=> Time
irb(main):033:0> Mongoid.use_activesupport_time_zone?
=> true
```